src/storagemanager.cpp

```diff
@@ -138,19 +138,17 @@ void Storage::ResetSettings()
 
 void Storage::setProfile(const uint32_t profileNum)
 {
-	if (profileNum < 1 || profileNum > 4) return;
-	setFunctionalPinMappings(profileNum);
-	this->config.gamepadOptions.profileNumber = profileNum;
+	this->config.gamepadOptions.profileNumber = (profileNum < 1 || profileNum > 4) ? 1 : profileNum;
 }
 
-void Storage::setFunctionalPinMappings(const uint32_t profileNum)
+void Storage::setFunctionalPinMappings()
 {
 	for (Pin_t pin = 0; pin < (Pin_t)NUM_BANK0_GPIOS; pin++) {
 		functionalPinMappings[pin] = this->config.gpioMappings.pins[pin].action;
 	}
-	if (profileNum < 2 || profileNum > 4) return;
+	if (config.gamepadOptions.profileNumber < 2 || config.gamepadOptions.profileNumber > 4) return;
```

Comment:
Can this ever happen? or should it be if (config.gamepadOptions.profileNumber ==1) return; since we have checks earlier?

Reply:
Just a sanity check since we're doing array stuff. Shouldn't happen but I guess if anyone did a weird POST to the API or edited the protobuf directly it could be a bad value at startup.

Reply:
Ahhh got it!